### PR TITLE
Added unit tests for NotifyAdminForTaskSignupStatusChangeHandler

### LINKED ISSUE #861: write unit tests for NotifyAdminForTaskSignupStatusChange
@tonysurma please label with "jump-in". Part of Issue #620

### PR DESCRIPTION
**Pull request by ExoltenOne**
Added unit tests for NotifyAdminForTaskSignupStatusChangeHandler and changed a subject's format to a user mail when a first name and a last name of a user are empty.

Closes #861

### CODE REVIEW

**Comment by mgmccarthy** [comment]
@stevegordon, I'm not back until tomorrow night. Can u take a crack at the code review for this one or do you need me to pick it up?

**Comment by stevejgordon** [comment]
@mgmccarthy I should be able to take a look at this today during the code-a-thon

**Comment by stevejgordon** [comment]
@ExoltenOne - Sorry didn't look at this yet. Can try later unless @mgmccarthy is able to run his eyes over it first!

**Comment by ExoltenOne** [comment]
@stevejgordon don't worry. No need to rush, but I admit I'll be happy to see my first pull request merged 😄

**Comment by ExoltenOne** [comment]
I see your point and I will change it to the previous version as you requested. I must agree that relying on the validation is an obvious choice, but I used secure code apart from this Yeah but maybe this is a consequence of dealing with finance topics on my daily basis.

**Comment by ExoltenOne** [comment]
@mgmccarthy I just pushed the original version.

File: AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForTaskSignupStatusChangeHandler.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public NotifyAdminForTaskSignupStatusChangeHandler(AllReadyContext context, IMed
             {
                 var link = $"View event: http://{_options.Value.SiteBaseUrl}/Admin/Task/Details/{taskSignup.Task.Id}";
 
-                var subject = !string.IsNullOrEmpty(volunteer.FirstName) && !string.IsNullOrEmpty(volunteer.LastName)
+                var subject = volunteer.FirstName != null && volunteer.LastName != null
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForTaskSignupStatusChangeHandler.cs]: @ExoltenOne, what I'm trying to say is we don't need to check for null. It's impossible for FirstName or LastName to be null b/c of how users are created in the system. This line should be deleted and this is how the subject should get assigned to the Subject property on the NotifyVolunteersViewModel:

```
var command = new NotifyVolunteersCommand
{
    ViewModel = new NotifyVolunteersViewModel
    {
        EmailMessage = message,
        HtmlMessage = message,
        EmailRecipients = new List<string> { adminEmail },
        Subject = $"{volunteer.FirstName} {volunteer.LastName}" 
    }
};
```

Comment (ExoltenOne) [inline AllReadyApp/Web-App/AllReady/Features/Notifications/NotifyAdminForTaskSignupStatusChangeHandler.cs]: @mgmccarthy ok, I thought we should go back to what was initially. I will change it in a minute.

**Comment by ExoltenOne** [comment]
@mgmccarthy This time should be better and sorry for the misunderstanding.

**Comment by mgmccarthy** [comment]
Thanks for the work on this @ExoltenOne. @tonysurma :shipit:

**Pull request merged**